modules/feedadBidAdapter.js

```diff
@@ -0,0 +1,291 @@
+ * @return {FeedAdApiBidRequest}
+ */
+function createApiBidRParams(request) {
+  return {
+    ad_type: 0,
+    client_token: request.params.clientToken,
+    placement_id: request.params.placementId,
+    sdk_version: `prebid_${VERSION}`,
+    app_hybrid: false,
+  };
+}
+
+/**
+ * Builds the bid request to the FeedAd Server
+ * @param {BidRequest[]} validBidRequests - all validated bid requests
+ * @param {object} bidderRequest - meta information
+ * @return {ServerRequest|ServerRequest[]}
+ */
+function buildRequests(validBidRequests, bidderRequest) {
+  if (!bidderRequest) {
+    return [];
+  }
+  let acceptableRequests = validBidRequests.filter(request => !isMediaTypesEmpty(filterSupportedMediaTypes(request.mediaTypes)));
+  if (acceptableRequests.length === 0) {
+    return [];
+  }
+  let data = Object.assign({}, bidderRequest, {
+    bids: acceptableRequests.map(req => {
+      req.params = createApiBidRParams(req);
+      return req;
+    })
+  });
+  data.bids.forEach(bid => BID_METADATA[bid.bidId] = {
+    referer: data.refererInfo.referer,
+    transactionId: bid.transactionId
+  });
+  return {
+    method: 'POST',
+    url: `${API_ENDPOINT}${API_PATH_BID_REQUEST}`,
+    data,
+    options: {
+      contentType: 'application/json'
+    }
+  };
+}
+
+/**
+ * Adapts the FeedAd server response to Prebid format
+ * @param {ServerResponse} serverResponse - the FeedAd server response
+ * @param {BidRequest} request - the initial bid request
+ * @returns {Bid[]} the FeedAd bids
+ */
+function interpretResponse(serverResponse, request) {
+  /**
+   * @type FeedAdApiBidResponse[]
+   */
+  return typeof serverResponse.body === 'string' ? JSON.parse(serverResponse.body) : serverResponse.body;
+}
+
+/**
+ * Creates the parameters for the FeedAd tracking call
+ * @param {object} data - prebid data
+ * @param {'prebid_bidWon'|'prebid_bidTimeout'} klass - type of tracking call
+ * @return {FeedAdApiTrackingParams|null}
+ */
+function createTrackingParams(data, klass) {
+  const bidId = data.bidId || data.requestId;
+  if (!BID_METADATA.hasOwnProperty(bidId)) {
+    return null;
+  }
+  const {referer, transactionId} = BID_METADATA[bidId];
+  delete BID_METADATA[bidId];
+  return {
+    app_hybrid: false,
+    client_token: data.params[0].clientToken,
+    placement_id: data.params[0].placementId,
+    klass,
+    prebid_auction_id: data.auctionId,
+    prebid_bid_id: bidId,
+    prebid_transaction_id: transactionId,
+    referer,
+    sdk_version: VERSION
+  };
+}
+
+/**
+ * Creates a tracking handler for the given event type
+ * @param klass - the event type
+ * @return {Function} the tracking handler function
+ */
+function trackingHandlerFactory(klass) {
+  return (data, xhr) => {
+    if (!data) {
+      return;
+    }
+    let params = createTrackingParams(data, klass);
+    if (params) {
+      xhr = typeof xhr === 'function' ? xhr : ajax; // required to test calls to the ajax method because it cannot be mocked
```

Comment:
There are other ways to check the request made by an ajax request.  You can use the `sinon.useFakeXMLHttpRequest()` that would make a handler for the requests and manage the out-going request objects to inspect them for the test.

The following link can give some idea of how this could work:
https://github.com/prebid/Prebid.js/blob/master/test/spec/AnalyticsAdapter_spec.js#L28